### PR TITLE
DataSearchWidget: Add aria-label for better accessibility

### PR DESCRIPTION
**Pull request by apepper**

### CODE REVIEW

File: src/Widgets/DataSearchWidget/DataSearchWidgetComponent.tsx [hunk 1]
```diff
@@ -43,7 +44,11 @@ provideComponent(DataSearchWidget, ({ widget }) => {
         defaultValue={search}
       />
       <span className="input-group-btn">
-        <button className={`btn ${buttonColor}`} type="submit">
+        <button
+          className={`btn ${buttonColor}`}
+          type="submit"
+          aria-label={localizeSearchInputLabel()}
```
Comment (dcsaszar) [inline src/Widgets/DataSearchWidget/DataSearchWidgetComponent.tsx:50]: * [ ] FYI looking at the current values for `placeholder` (typically "Search for X") we could repurpose these and use them for both the placeholder and the button label. If we do so, it should be made clear in the editing config though.

**Review by dcsaszar — APPROVED** [review]
See idea/comment.

**Comment by apepper** [comment]
REVIEW ([bf298f7](https://github.com/Scrivito/scrivito-portal-app/pull/636/commits/bf298f72b382fcc0e8a2ac47592132e4d998e7fa))

File: src/Widgets/DataSearchWidget/DataSearchWidgetEditingConfig.ts [hunk 2]
```diff
@@ -17,7 +17,11 @@ provideEditingConfig(DataSearchWidget, {
         { value: 'btn-secondary', title: 'Secondary color' },
       ],
     },
-    placeholder: { title: 'Placeholder text' },
+    placeholder: {
+      title: 'Placeholder text',
+      description:
+        'This value is also utilized for the aria label of the search button.',
```
Comment (dcsaszar) [inline src/Widgets/DataSearchWidget/DataSearchWidgetEditingConfig.ts:23]: * [ ] FYI ARIA is uppercase (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA)

**Review by dcsaszar — APPROVED** [review]

**Pull request merged**